client/blocks/reader-post-card/index.jsx

```diff
@@ -158,15 +161,33 @@ export default class ReaderPostCard extends React.Component {
 				</StandardPost>;
 		}
 
+		// set up post byline
+		let postByline;
+
+		if ( isDiscoverStream && ! isEmpty( discoverPick ) ) {
+			// create a post like object with some props from the discover post
```

Comment:
Sounds like a plan!
This PR is blocking a few next steps so I'd like to ship it soon. I'm all for normalizing this but would like to do that in a new PR.